crud/program/src/state.rs

```diff
@@ -0,0 +1,87 @@
+//! Program state
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{program_pack::IsInitialized, pubkey::Pubkey},
+};
+
+/// Struct wrapping data and providing metadata
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub struct AccountData {
+    /// Struct version, allows for upgrades to the program
+    pub version: u8,
```

Comment:
How do you imagine `version` being used?  For now it would always be set to 1 on Initialize?

Reply:
The concept is that a program developer iterates on a record, they might need to modify the record type by adding / removing fields, so the version allows the program to handle all types

Reply:
eventually there could also be an "UpdateVersion" instruction that would take a v1 and turn it into v2, and then Initialize would set new accounts to v2

Reply:
I feel like I'm missing an important piece of the motivation behind this program, as it seems to me that versioning of the data should be a client concern. 

"UpdateVersion" could be either an atomic `CloseAccount`, `Initialize`, `Write`, or once we support account resizing (with a new version of this program deployed) could be an atomic `Resize` (new instruction) and `Write`.

Reply:
I see what you mean -- that certainly works too.  No matter what, we need to know if the account data is initialized with some field at the start.  By having that field as `version` instead of just `is_initialized`, it can carry additional information, to be used by the client or not.  If you think it's clearer to change to an `is_initialized` field, I don't have any objections

Reply:
Got it, seems ok to me as is then :)